infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -902,27 +919,12 @@ private LogEntry getLogEntry(long address, LogData entry) {
      * @param buf     the buffer to write
      * @throws IOException IO exception
      */
-    private static void safeWrite(FileChannel channel, ByteBuffer buf) throws IOException {
-        long prev = channel.position();
-        try {
-            do {
-                channel.write(buf);
-            } while (buf.hasRemaining());
-
-        } catch (IOException e) {
-            // Write failed restore the channels position, so the subsequent writes
-            // can overwrite the failed write.
-
-            // Note that after rewinding the channel pointer, it is important to truncate
-            // any bytes that were written. This is required to avoid an ambiguous case
-            // where a subsequent write (after a failed write) succeeds but writes less
-            // bytes than the partially written buffer. In that case, the log unit can't
-            // determine if the bytes correspond to a partially written buffer that needs
-            // to be ignored, or if the bytes correspond to a corrupted metadata field.
-            channel.position(prev);
-            channel.truncate(prev);
-            channel.force(true);
-            throw e;
+    private void writeByteBuffer(FileChannel channel, ByteBuffer buf) throws IOException {
+        // On IOExceptions this class should be reinitialized, so consuming
+        // the buffer size and failing on the write should be an issue
```

Comment:
did you mean - "shouldn't be an issue"?

Reply:
No, it should.

Reply:
If an IOException happens on a write and we retry, we have to throw away that FD, because it could be the case that the subsequent fsync wouldn't see the exception and "fail silently".